pkgs/applications/networking/browsers/firefox/wrapper.nix

```
@@ -102,18 +102,15 @@ let
 
       nameArray = builtins.map(a: a.name) (lib.optionals usesNixExtensions nixExtensions);
 
-      requiresSigning = browser ? MOZ_REQUIRE_SIGNING
-                     -> toString browser.MOZ_REQUIRE_SIGNING != "";
-
       # Check that every extension has a unqiue .name attribute
       # and an extid attribute
       extensions = if nameArray != (lib.unique nameArray) then
         throw "Firefox addon name needs to be unique"
-      else if requiresSigning && !lib.hasSuffix "esr" browser.name then
-        throw "Nix addons are only supported without signature enforcement (eg. Firefox ESR)"
+      else if browser.requireSigning && browser.allowAddonSideload then
```

This should throw when either requireSigning is true or allowAddonSideload is false currently is only throws when both are true

```suggestion
      else if browser.requireSigning || !browser.allowAddonSideload then
```

---

Good catch, I don't think I would have checked and realized that. Fixed.